slasher/rpc/server.go

```diff
@@ -91,5 +91,37 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
 // IsSlashableBlock returns an proposer slashing if the block submitted
 // is a double proposal.
 func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconBlockHeader) (*slashpb.ProposerSlashingResponse, error) {
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "detection.IsSlashableBlock")
+	defer span.End()
+
+	gvr, err := ss.beaconClient.GenesisValidatorsRoot(ctx)
+	if err != nil {
+		return nil, err
+	}
+	blockEpoch := helpers.SlotToEpoch(req.Header.Slot)
+	fork, err := p2putils.Fork(blockEpoch)
+	if err != nil {
+		return nil, err
+	}
+	domain, err := helpers.Domain(fork, blockEpoch, params.BeaconConfig().DomainBeaconProposer, gvr)
+	if err != nil {
+		return nil, err
+	}
+	pkMap, err := ss.beaconClient.FindOrGetPublicKeys(ctx, []uint64{req.Header.ProposerIndex})
```

Comment:
This result can likely also use a caching-layer given there may be many repeat requests

Reply:
This method is cached